logstash-core/lib/logstash/persisted_queue_config_validator.rb

```diff
@@ -47,8 +47,9 @@ def check(running_pipelines, pipeline_configs)
         max_bytes = config.settings.get("queue.max_bytes").to_i
         page_capacity = config.settings.get("queue.page_capacity").to_i
         pipeline_id = config.settings.get("pipeline.id")
-        queue_path = config.settings.get("path.queue")
-        pq_page_glob = ::File.join(queue_path, pipeline_id, "page.*")
+        queue_path = ::File.join(config.settings.get("path.queue"), pipeline_id)
+        pq_page_glob = ::File.join(queue_path, "page.*")
+        create_dirs(queue_path)
```

Comment:
@mashhurs It works well with mounting the filesystem to the queue folder.
I have tired to use symlink pointing to path.queue/pipeline_id
It fails with with FileAlreadyExistsException in create_dirs() and [QueueFactory](https://github.com/elastic/logstash/blob/v7.17.0/logstash-core/src/main/java/org/logstash/ackedqueue/QueueFactoryExt.java#L60)

I think we can track the problem in a new issue if we consider symlinks is a valid scenario. WDYT @jsvd ?

Reply:
Thanks @kaisecheng , your setting is also valid and as we took offline to test together and share thoughts, I have added exception handling logic.